lib_xud/src/core/included/XUD_Token_Setup_DI.S

```diff
@@ -24,24 +24,25 @@ ClearStall:                                     // CRC OK
     ldaw       r11, dp[handshakeTable_OUT]      // Note, we can speed this up by assuming SETUP only received on EP 0
     ldc        r6, USB_PIDn_NAK
     stw        r6, r11[r10]
-    ldaw       r11, dp[handshakeTable_IN]
+    {ldw       r11, sp[STACK_HANDSHAKETABLEIN]
+    ldc        r1, 0}
     stw        r6, r11[r10]
+    ldaw       r11, dp[g_stallTable_IN]
+    stw        r1, r11[r10]
 
 SendSetupAck:
     ldc        r11, USB_PIDn_ACK
     outpw      res[TXD], r11, 8
 
 StoreTailData:                                  // TODO: don't assume setups are 8 bytes + crc
-    ldc        r11, 0
-    stw        r11, r5[r10]                     // Clear ready
+    stw        r1, r5[r10]                      // Clear ready
```

Comment:
Does anything branch to `StoreTailData`?  I do not see a branch to it in this file, but i know that some pieces of code are `#include`d by others.  If a branch to `StoreTailData` does exist, is there a guarantee that r1 has zero in it?

Reply:
There are no branches to it but the code falls to it from above.
I tend to use a fair amount of labels when I'm coding asm for my sanity, readability and also debugging - easier to see when the code is when breaking.

Reply:
I wondered if that was the case.  No worries then.